src/compiler/binder.ts

```diff
@@ -2608,10 +2619,15 @@ namespace ts {
 
         // If a FunctionExpression contains an asterisk token, or its subtree has marked the container
         // as needing to capture the lexical this, then this node is ES6 syntax.
-        if (asteriskToken || (subtreeFlags & TransformFlags.ES6FunctionSyntaxMask)) {
+        if (subtreeFlags & TransformFlags.ES6FunctionSyntaxMask) {
             transformFlags |= TransformFlags.AssertES6;
         }
 
+        // Currently, we only support generators that were originally async function bodies.
+        if (asteriskToken && node.emitFlags & NodeEmitFlags.AsyncFunctionBody) {

```

Comment:
how can this `node.emitFlags & NodeEmitFlags.AsyncFunctionBody` be observed here since the flag is set in the transformer's factory.

Also what happen to the case of `asteriskToken` but not AsyncFunctionBody?